src/v/cluster/types.h

```diff
@@ -1369,13 +1368,6 @@ struct topic_configuration
         properties = read_nested<topic_properties>(in, h._bytes_left_limit);
 
         if (h._version < 1) {
-            maybe_adjust_retention_policies(
-              properties.shadow_indexing,
-              properties.retention_bytes,
-              properties.retention_duration,
-              properties.retention_local_target_bytes,
-              properties.retention_local_target_ms);
-
             // Legacy tiered storage topics do not delete data on
             // topic deletion.
             properties.remote_delete
```

Comment:
I ran into some issues with the compat tests -- I think this is harmless to leave in place for now (whereas the retention bits had to go in order to avoid double-swapping).

Reply:
Urgh... Those compat tests haunt my dreams. I agree that is should be harmless to leave this here.